qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -0,0 +1,480 @@
+            self.drawer.set_options(**draw_options)
+
+        super().set_options(**fields)
+
+    @abstractmethod
+    def _generate_fit_guesses(
+        self,
+        user_opt: FitOptions,
+        curve_data: CurveData,
+    ) -> Union[FitOptions, List[FitOptions]]:
+        """Create algorithmic guess with analysis options and curve data.
+
+        Args:
+            user_opt: Fit options filled with user provided guess and bounds.
+            curve_data: Formatted data collection to fit.
+
+        Returns:
+            List of fit options that are passed to the fitter function.
+        """
+
+    def _format_data(
+        self,
+        curve_data: CurveData,
+    ) -> CurveData:
+        """Post-processing for fit data collection.
+
+        Args:
+            curve_data: Raw data collection created from experiment results.
+
+        Returns:
+            Formatted data.
+        """
+        # take average over the same x value by keeping sigma
+        data_allocation, xdata, ydata, sigma, shots = multi_mean_xy_data(
+            series=curve_data.data_allocation,
+            xdata=curve_data.x,
+            ydata=curve_data.y,
+            sigma=curve_data.y_err,
+            shots=curve_data.shots,
+            method="shots_weighted",
+        )
+
+        # sort by x value in ascending order
+        data_allocation, xdata, ydata, sigma, shots = data_sort(
+            series=data_allocation,
+            xdata=xdata,
+            ydata=ydata,
+            sigma=sigma,
+            shots=shots,
+        )
+
+        return CurveData(
+            x=xdata,
+            y=ydata,
+            y_err=sigma,
+            shots=shots,
+            data_allocation=data_allocation,
+            labels=curve_data.labels,
+        )
+
+    def _evaluate_quality(
+        self,
+        fit_data: FitData,
+    ) -> Union[str, None]:
+        """Evaluate quality of the fit result.
+
+        Args:
+            fit_data: Fit outcome.
+
+        Returns:
+            String that represents fit result quality. Usually "good" or "bad".
+        """
+        if fit_data.reduced_chisq < 3.0:
+            return "good"
+        return "bad"
+
+    def _run_data_processing(
+        self,
+        raw_data: List[Dict],
+        series: List[SeriesDef],
+    ) -> CurveData:
+        """Perform data processing from the experiment result payload.
+
+        Args:
+            raw_data: Payload in the experiment data.
+            series: List of series definition defining filtering condition.
+
+        Returns:
+            Un-formatted data collection.
+        """
+        x_key = self.options.x_key
+
+        try:
+            xdata = np.asarray([datum["metadata"][x_key] for datum in data], dtype=float)
+        except KeyError as ex:
+            raise DataProcessorError(
+                f"X value key {x_key} is not defined in circuit metadata."
+            ) from ex
+
+        ydata = self.options.data_processor(data)
```

Comment:
Option is automatically overridden in `_preparation` method that is newly introduced. But we are going to remove `get_processor` method there and use `_finalize` method instead to directly set option value from experiment (this will solve run option mismatch issue we have now), once `DataProcessor` becomes a centric class for data processing (currently several experiments don't use processor).